### ISSUE TITLE
[share_plus] iOS 12 and lower crashes at launch due to missing LinkPresentation.framework

### ISSUE THREAD

**Issue by JagandeepBrar** [fluttercommunity/plus_plugins#222]
## System info

Issue occurs on: iOS
Plugin name: share_plus
Plugin version: 2.0.1
Flutter doctor output:

```
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.3 20E5231a darwin-x64, locale en-CA)
    • Flutter version 2.0.5 at /Users/jagandeepbrar/Development/flutter
    • Framework revision adc687823a (7 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/jagandeepbrar/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (2 available)
    • iPhone 6 (mobile) • 30F1098D-66DB-41FA-ACA9-A06D5FD31AAB • ios        • com.apple.CoreSimulator.SimRuntime.iOS-12-4 (simulator)
    • macOS (desktop)   • macos                                • darwin-x64 • macOS 11.3 20E5231a darwin-x64

• No issues found!
```

## Steps to Reproduce

1. Create a new project and add the `share_plus` package
2. Build and run the project on a device running iOS 12.x or lower
3. The application will crash at startup

## Logs

Using `flutter run --verbose`, the logs stop without any error at "attaching to observatory..." but when running the project through XCode, I get the following errors:

0 __abort_with_payload
```
dyld`__abort_with_payload:
    0x10fcbfae0 <+0>:  movl   $0x2000209, %eax          ; imm = 0x2000209 
    0x10fcbfae5 <+5>:  movq   %rcx, %r10
    0x10fcbfae8 <+8>:  syscall 
->  0x10fcbfaea <+10>: jae    0x10fcbfaf4               ; <+20>
    0x10fcbfaec <+12>: movq   %rax, %rdi
    0x10fcbfaef <+15>: jmp    0x10fcbdcf0               ; cerror_nocancel
    0x10fcbfaf4 <+20>: retq   
    0x10fcbfaf5 <+21>: nop    
    0x10fcbfaf6 <+22>: nop    
    0x10fcbfaf7 <+23>: nop    
```

10 _dyld_start
```
dyld`_dyld_start:
    0x10fc64000 <+0>:  popq   %rdi
    0x10fc64001 <+1>:  pushq  $0x0
    0x10fc64003 <+3>:  movq   %rsp, %rbp
    0x10fc64006 <+6>:  andq   $-0x10, %rsp
    0x10fc6400a <+10>: subq   $0x10, %rsp
    0x10fc6400e <+14>: movl   0x8(%rbp), %esi
    0x10fc64011 <+17>: leaq   0x10(%rbp), %rdx
    0x10fc64015 <+21>: leaq   -0x101c(%rip), %rcx
    0x10fc6401c <+28>: leaq   -0x8(%rbp), %r8
    0x10fc64020 <+32>: callq  0x10fc64062               ; dyldbootstrap::start(dyld3::MachOLoaded const*, int, char const**, dyld3::MachOLoaded const*, unsigned long*)
->  0x10fc64025 <+37>: movq   -0x8(%rbp), %rdi
    0x10fc64029 <+41>: cmpq   $0x0, %rdi
    0x10fc6402d <+45>: jne    0x10fc6403f               ; <+63>
    0x10fc6402f <+47>: movq   %rbp, %rsp
    0x10fc64032 <+50>: addq   $0x8, %rsp
    0x10fc64036 <+54>: movq   $0x0, %rbp
    0x10fc6403d <+61>: jmpq   *%rax
    0x10fc6403f <+63>: addq   $0x10, %rsp
    0x10fc64043 <+67>: pushq  %rdi
    0x10fc64044 <+68>: movq   0x8(%rbp), %rdi
    0x10fc64048 <+72>: leaq   0x10(%rbp), %rsi
    0x10fc6404c <+76>: leaq   0x8(%rsi,%rdi,8), %rdx
    0x10fc64051 <+81>: movq   %rdx, %rcx
    0x10fc64054 <+84>: movq   (%rcx), %r8
    0x10fc64057 <+87>: addq   $0x8, %rcx
    0x10fc6405b <+91>: testq  %r8, %r8
    0x10fc6405e <+94>: jne    0x10fc64054               ; <+84>
    0x10fc64060 <+96>: jmpq   *%rax
```

```
dyld: Library not loaded: /System/Library/Frameworks/LinkPresentation.framework/LinkPresentation
  Referenced from: /Users/jagandeepbrar/Library/Developer/CoreSimulator/Devices/30F1098D-66DB-41FA-ACA9-A06D5FD31AAB/data/Containers/Bundle/Application/8254F4F4-DB7A-4521-8D45-ED25B77CFEE6/Runner.app/Runner
  Reason: image not found
dyld: launch, loading dependent libraries
DYLD_FRAMEWORK_PATH=/Users/jagandeepbrar/Library/Developer/Xcode/DerivedData/Runner-evzawmfibposwygaviyermwngfvp/Build/Products/Debug-iphonesimulator
DYLD_FALLBACK_LIBRARY_PATH=/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib
DYLD_ROOT_PATH=/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.4.simruntime/Contents/Resources/RuntimeRoot
DYLD_FALLBACK_FRAMEWORK_PATH=/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.4.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks
DYLD_INSERT_LIBRARIES=/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libBacktraceRecording.dylib:/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.4.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libMainThreadChecker.dylib:/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.4.simruntime/Contents/Resources/RuntimeRoot/Developer/Library/Pri
```

## Additional Context

[LinkPresentation.framework](https://developer.apple.com/documentation/linkpresentation) was added in iOS 13.0, and is not available for devices running older versions.

`share_plus` 2.0.1, or more specifically [this pull request](https://github.com/fluttercommunity/plus_plugins/pull/207), is when `LinkPresentation.framework` started to be integrated into the package.

Can confirm that rolling back to `share_plus` 2.0.0 solves this, and the application no longer crashes on iOS 12 or lower.

**Comment by miquelbeltran** [comment]
Hi @JagandeepBrar thanks for the report.

In #207 the author said that "Verified working on iOS 14.4, 12.0 simulators". Can you provide any hints @ateich ?

**Comment by ateich** [comment]
I haven't been able to reproduce the issue in the iOS Simulator (iPhone 5s, iOS 12.0)

I'm charging my old iPhone 5s so I can test on that next

I think the problem may be in the build target's settings, specifically it has an other linker flag that is currently `-framework "LinkPresentation"` instead of `-weak_framework "LinkPresentation"`

I'll test it out on the device as soon as I can

**Comment by JagandeepBrar** [comment]
> I haven't been able to reproduce the issue in the iOS Simulator (iPhone 5s, iOS 12.0)

Interesting, I just tested the following (all simulators, I do not have any physical devices < iOS 13):
- iOS 12.4 / iPhone 6
- iOS 12.0 / iPhone 6
- iOS 11.4 / iPhone 6

On all instances, the application crashes on startup as per the original issue. Only once using 13.0 or higher does the application launch successfully.

**Comment by miquelbeltran** [comment]
Version `2.0.2` has been released with the changes from @ateich on #224 

Please @JagandeepBrar run a `flutter pub upgrade`, check that version `2.0.2` is used and let us know if the fix worked.

**Comment by JagandeepBrar** [comment]
Confirmed fixed! Tested 2.0.1 on iOS 12.4: crash. 2.0.2 no longer crashes and can verify the share sheet displays and works as expected.

**Comment by miquelbeltran** [comment]
Thanks a lot @ateich for providing the fix!
